lib/ts/recipe/emailpassword/recipe.ts

```diff
@@ -101,6 +105,63 @@ export default class Recipe extends RecipeModule {
                 Recipe.instance = new Recipe(Recipe.RECIPE_ID, appInfo, isInServerlessEnv, config, {
                     emailDelivery: undefined,
                 });
+
+                PostSuperTokensInitCallbacks.addPostInitCallback(() => {
+                    const mfaInstance = MultiFactorAuthRecipe.getInstance();
+                    if (mfaInstance !== undefined) {
+                        mfaInstance.addGetAllFactorsFromOtherRecipesFunc((tenantConfig) => {
+                            if (tenantConfig.emailPassword.enabled === false) {
+                                return [];
+                            }
+                            return ["emailpassword"];
+                        });
+                        mfaInstance.addGetFactorsSetupForUserFromOtherRecipes(async (user: User) => {
+                            for (const loginMethod of user.loginMethods) {
+                                // We deliberately do not check for matching tenantId because we assume
+                                // MFA is app-wide by default. User can always override MFA function
+                                // to make it tenant specific.
+                                if (loginMethod.recipeId === Recipe.RECIPE_ID) {
+                                    return ["emailpassword"];
+                                }
+                            }
+                            return [];
+                        });
+                        mfaInstance.addGetEmailsForFactorFromOtherRecipes((user: User, sessionRecipeUserId) => {
```

Comment:
this will have to be tenant specific cause if it's not, it may lead to a case where we pass in the email from another tenant, and then if another tenant uses that email + password to sign in, it won't work cause signIn is per tenant.

This causes other issues where if a user has otp-email configured on a user level, and they login with a tenant that doesn't have passwordless account linked to them, they will be asked to enter a new email if the first factor is a phone number for that tenant. Even though this user has an email passwordless user in another tenant. This is strange...

Reply:
done